tests/cli/functional/actions/test_select_client_account.py

```diff
@@ -0,0 +1,191 @@
+import click
+import pytest
+from eth_utils import is_checksum_address
+from unittest.mock import Mock
+from web3 import Web3
+
+from nucypher.blockchain.eth.actors import Wallet
+from nucypher.blockchain.eth.clients import EthereumClient
+from nucypher.blockchain.eth.interfaces import BlockchainInterfaceFactory
+from nucypher.blockchain.eth.token import NU
+from nucypher.cli.actions.select import select_client_account
+from nucypher.cli.literature import NO_ETH_ACCOUNTS
+from nucypher.config.constants import TEMPORARY_DOMAIN
+from tests.constants import MOCK_PROVIDER_URI, MOCK_SIGNER_URI, NUMBER_OF_ETH_TEST_ACCOUNTS
+
+
+@pytest.mark.parametrize('selection', range(NUMBER_OF_ETH_TEST_ACCOUNTS))
+def test_select_client_account(mock_click_prompt, test_emitter, mock_testerchain, selection):
+    """Fine-grained assertions about the return value of interactive client account selection"""
+    mock_click_prompt.return_value = selection
+    expected_account = mock_testerchain.client.accounts[selection]
+    selected_account = select_client_account(emitter=test_emitter, provider_uri=MOCK_PROVIDER_URI)
+    assert selected_account, "Account selection returned Falsy instead of an address"
+    assert isinstance(selected_account, str), "Selection is not a str"
+    assert is_checksum_address(selected_account), "Selection is not a valid checksum address"
+    assert selected_account == expected_account, "Selection returned the wrong address"
+
+
+def test_select_client_account_with_no_accounts(mocker,
+                                                mock_click_prompt,
+                                                test_emitter,
+                                                mock_testerchain,
+                                                stdout_trap):
+    mocker.patch.object(EthereumClient, 'accounts', return_value=[])
+    with pytest.raises(click.Abort):
+        select_client_account(emitter=test_emitter, provider_uri=MOCK_PROVIDER_URI)
+    output = stdout_trap.getvalue()
+    assert NO_ETH_ACCOUNTS in output
+
+
+def test_select_client_account_ambiguous_source(mock_click_prompt, test_emitter, mock_testerchain):
+
+    #
+    # Implicit wallet
+    #
+
+    error_message = "At least a provider URI or signer URI is necessary to select an account"
+    with pytest.raises(ValueError, match=error_message):
+        select_client_account(emitter=test_emitter)
+
+    error_message = "Pass either signer or signer_uri but not both."
+    with pytest.raises(ValueError, match=error_message):
+        select_client_account(emitter=test_emitter, signer=Mock(), signer_uri=MOCK_SIGNER_URI)
+
+    #
+    # Explicit wallet
+    #
+
+    error_message = "If pass a wallet, don't pass a signer, provider URI, or signer URI also."
```

Comment:
Based on my previous suggestion:

```suggestion
    error_message = "If a wallet is provided, don't provide a signer, provider URI, or signer URI."
```

Reply:
:heavy_check_mark: